math/dec_bench_test.go

```diff
@@ -0,0 +1,95 @@
+package math
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkCompareLegacyDecAndNewDec(b *testing.B) {
+	legacyB1 := LegacyNewDec(100)
+	legacyB2 := LegacyNewDec(5)
+	newB1 := NewDecFromInt64(100)
+	newB2 := NewDecFromInt64(5)
+
+	b.Run("LegacyDec", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_ = legacyB1.Quo(legacyB2)
+		}
+	})
+
+	b.Run("NewDec", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, _ = newB1.Quo(newB2)
+		}
+	})
+}
+
+func BenchmarkCompareLegacyDecAndNewDecQuoInteger(b *testing.B) {
+	legacyB1 := LegacyNewDec(100)
+	newB1 := NewDecFromInt64(100)
+
+	b.Run("LegacyDec", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_ = legacyB1.Quo(LegacyNewDec(1))
+		}
+	})
+
+	b.Run("NewDec", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, _ = newB1.QuoInteger(NewDecFromInt64(1))
+		}
+	})
```

Comment:
These benchmarks are wrong, they benchmark the dec allocation!

But to do a useful benchmark, we need numbers at far bigger word sizes, not single word!